house/house.go

```diff
@@ -277,24 +340,75 @@ func (s *Server) Address(ctx context.Context, in *pb.AddressRequest) (*pb.Reply,
 	incoming := in
 	newDevice := true
 	for _, houseDevice := range houseDevices {
-		if houseDevice.Id.Ip == incoming.Ip && houseDevice.Id.Mac == incoming.Mac ||
-			houseDevice.Id.Mac == incoming.Mac && incoming.Mac != "" ||
-			houseDevice.Id.Ip == incoming.Ip && incoming.Mac == "" ||
-			houseDevice.Id.Ip != incoming.Ip && incoming.Mac != "" && incoming.Mac == houseDevice.Id.Mac ||
-			houseDevice.Id.Ip == incoming.Ip && incoming.Ip != "" && incoming.Mac != houseDevice.Id.Mac {
+		if houseDevice.Id.UUID == incoming.Mac ||
+			houseDevice.Id.UUID == incoming.Ip {
+			//if houseDevice.Id.Ip == incoming.Ip && houseDevice.Id.Mac == incoming.Mac ||
+			//	houseDevice.Id.Mac == incoming.Mac && incoming.Mac != "" ||
+			//	houseDevice.Id.Ip == incoming.Ip && incoming.Mac == "" ||
+			//	houseDevice.Id.Ip != incoming.Ip && incoming.Mac != "" && incoming.Mac == houseDevice.Id.Mac ||
+			//	houseDevice.Id.Ip == incoming.Ip && incoming.Ip != "" && incoming.Mac != houseDevice.Id.Mac {
 			newDevice = false
 			err := s.existingDevice(houseDevice, incoming)
 			if err != nil {
 				return nil, err
 			}
+			s.iotStatusManager()
+
 		}
 	}
+
 	if newDevice {
 		err := s.newDevice(in)
 		if err != nil {
 			return nil, err
 		}
 	}
+
+	return &pb.Reply{Acknowledged: true}, nil
+}
+
+func (s *Server) Ack(ctx context.Context, in *pb.BleRequest) (*pb.Reply, error) {
```

Comment:
warning: exported method Server.Ack should have comment or be unexported (golint)